libs/core/libs/shared/src/document_repo.rs

```diff
@@ -81,23 +90,11 @@ pub struct DocActivityMetrics {
     pub write_count: StatisticValue,
 }
 
-impl Eq for DocActivityMetrics {}
-impl Ord for DocActivityMetrics {
-    fn cmp(&self, other: &Self) -> Ordering {
-        other.score().cmp(&self.score())
-    }
-}
-
-impl PartialOrd for DocActivityMetrics {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(other.score().cmp(&self.score()))
-    }
-}
-
 impl DocActivityMetrics {
-    pub fn score(&self) -> i64 {
-        let timestamp_weight = 70;
-        let io_count_weight = 30;
+    pub fn score(&self, weights: Option<RankingWeights>) -> i64 {
+        let weights = weights.unwrap_or(RankingWeights { temporality: 60, io: 40 });
```

Comment:
strikes me as an `unwrap_or_default` situation and then you can `impl Default for RankingWeights`